stellar-contract-env-host/src/host.rs

```diff
@@ -1549,30 +1549,131 @@ impl CheckedEnv for Host {
         Ok(self.to_host_obj(&sco)?.into())
     }
 
-    fn binary_copy_to_guest_mem(
+    fn binary_copy_to_linear_memory(
         &self,
         b: Object,
         i: RawVal,
         j: RawVal,
         l: RawVal,
     ) -> Result<RawVal, HostError> {
-        todo!()
+        //[offset in the host binary, pos in the linear memory, length of the slice to copy]
+        #[cfg(not(feature = "vm"))]
+        todo!();
+        #[cfg(feature = "vm")]
+        if let [i, j, l] = [i, j, l]
+            .iter()
+            .map(|v| u32::try_from(*v))
+            .collect::<Result<Vec<u32>, ConversionError>>()?
+            .as_slice()
+        {
+            self.visit_obj(b, move |hv: &Vec<u8>| {
+                let end_idx = i.checked_add(*l).ok_or(HostError::WithStatus(
```

Comment:
If you use `ok_or_else` you can lazily create the host error, rather than always creating the host error.
```suggestion
                let end_idx = i.checked_add(*l).ok_or_else(|| HostError::WithStatus(
```

Reply:
Thanks, I learned something :)